internal/config/config.go

```diff
@@ -439,9 +441,9 @@ func peerFromCR(p metallbv1beta2.BGPPeer, passwordSecrets map[string]corev1.Secr
 		return nil, fmt.Errorf("can not have both password and secret ref set in peer config %q/%q", p.Namespace, p.Name)
 	}
 
-	password := p.Spec.Password
+	secretPassword := ""
 	if p.Spec.PasswordSecret.Name != "" {
-		password, err = passwordFromSecretForPeer(p, passwordSecrets)
+		secretPassword, err = passwordFromSecretForPeer(p, passwordSecrets)
```

Comment:
w/ the other comment, I think it's cleaner to keep the "Password" field as "plain text password", not having the distinction on where it came from.